kasa/aestransport.py

```diff
@@ -123,17 +147,22 @@ async def send_secure_passthrough(self, request: str):
         }
         status_code, resp_dict = await self.client_post(url, json=passthrough_request)
         # _LOGGER.debug(f"secure_passthrough response is {status_code}: {resp_dict}")
-        if status_code == 200 and resp_dict["error_code"] == 0:
-            response = self._encryption_session.decrypt(  # type: ignore
-                resp_dict["result"]["response"].encode()
+
+        if status_code != 200:
+            raise SmartDeviceException(
+                f"Unable to send message for {self.host}, status code is {status_code}"
             )
-            _LOGGER.debug(f"decrypted secure_passthrough response is {response}")
-            resp_dict = json_loads(response)
-            return resp_dict
-        else:
-            self._handshake_done = False
-            self._login_token = None
-            raise AuthenticationException("Could not complete send")
+
+        self._handle_response_error_code(
+            resp_dict, "Error sending secure_passthrough message"
+        )
+
+        response = self._encryption_session.decrypt(  # type: ignore
+            resp_dict["result"]["response"].encode()
+        )
+        _LOGGER.debug(f"decrypted secure_passthrough response is {response}")
```

Comment:
Maybe we should remove this from the transport, as it's probably rather for the protocol class to log the full response payload?

Reply:
Done